basic/atm/atm_exercise.py

```diff
@@ -0,0 +1,198 @@
+
+    Args:
+        filename (str): The path to the atm details file.
+
+    Returns:
+         dict: Dictionary of all customer's data from the file.
+    """
+    with open(filename, "r") as atm_file:
+        customers = {}
+        for line in atm_file:
+            customer_id, the_pass, balance = line.split()
+            customer = {PASS_KEY: the_pass, BALANCE_KEY: float(balance)}
+            customers[customer_id] = customer
+    return customers
+
+
+def execute_choice(customers_data, customer_id, choice):
+    """Execute a function according to the choice given.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+        choice (str): The user's desired function to execute.
+    """
+    if choice == "1":
+        check_balance(customers_data, customer_id)
+
+    elif choice == "2":
+        cash_withdrawal(customers_data, customer_id)
+
+    elif choice == "3":
+        cash_deposit(customers_data, customer_id)
+
+    else:
+        change_password(customers_data, customer_id)
+
+
+def update_file(filename, data):
+    """Update the file with the data in the given dict 'data'.
+
+    Args:
+        filename (str): Path too the atm file.
+        data (dict): Dictionary with data from the atm file.
+    """
+    with open(filename, "w") as atm_file:
+        for key in data:
+            line = f"{key} {data[key][PASS_KEY]} {data[key][BALANCE_KEY]}\n"
+            atm_file.write(line)
+
+
+def operations_for_customer(data, customer_id):
+    """Get desired operation by user and execute it.
+
+    Args:
+        data (dict): Dictionary with data from the atm file.
+        customer_id (str): The specific customer id to work on.
+    """
+    msg = """What operation do you want to do?(-1 to stop)
+        1 to check balance
+        2 to withdrawal
+        3 for cash deposit
+        4 to change password\n"""
+    for operation in iter(partial(input, msg), "-1"):
+        if operation not in "1234":
+            print("you have to enter one of the options, try again.")
+            continue
+        execute_choice(data, customer_id, operation)
+
+
+def atm(filename):
+    """Perform operations on customers from the given file and update the data.
+
+    Args:
+        filename (str): The path to the file with the bank details.
+    """
+    data = get_data_from_file(filename)
+    msg = "enter the ID of the customer you want (-1 to turn off).\n"
+    for customer_id in iter(partial(input, msg), "-1"):
+        if customer_id not in data and customer_id != "-1":
+            err_msg = "The ID you entered doesn't exist, try again."
+            print(err_msg)
+            continue
+
+        operations_for_customer(data, customer_id)
+
+    update_file(filename, data)
+
+
+def main():
+    """Get file path and call atm function"""
+    try:
+        path = sys.argv[ATM_FILE]
```

Comment:
A little change to practice.
The bank manager want to see all clients balance, however, he don't know how to read the text file.
In order for the manager to read it, the data file should be in csv format.
Change your code to support csv data, instead of txt.